### PR TITLE
Implement simple post crud system

### LINKED ISSUE #9: Simple implementation of posts

### PR DESCRIPTION
**Pull request by bambamboole**
This PR implements simple post crud functionality

* Use spatie/laravel-sluggable for slugs on posts and tags
* Use vue-tags-input as base for own tag input component
* Add custom mutator to sync tags with post
* Refactor LoginController to not depend on the traits from laravel. Test randomly failed because trait was missing.

closes #9 #16

### CODE REVIEW

File: migrations/2020_03_07_181341_create_posts_table.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('users');
```
Comment (slashplusweb) [inline migrations/2020_03_07_181341_create_posts_table.php]: should be `Schema::dropIfExists('posts')`

File: resources/js/Pages/Posts/Edit.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<template>
+    <layout :title="'Edit: ' + post.title">
+
+        <div>
+
+            <div class="md:flex md:items-center md:justify-between">
+                <div class="flex-1 min-w-0">
+                    <h3 class="text-lg leading-6 font-medium text-gray-900">
```
Comment (slashplusweb) [inline resources/js/Pages/Posts/Edit.vue]: As described in Create.vue

File: resources/js/Pages/Posts/Index.vue [hunk 3]
```diff
@@ -1,18 +1,81 @@
 <template>
     <layout title="posts">
-        <h1>posts index</h1>
+        <div class="bg-white px-4 py-5 border-b border-gray-200 sm:px-6">
+            <div class="-ml-4 -mt-2 flex items-center justify-between flex-wrap sm:flex-no-wrap">
+                <div class="ml-4 mt-2">
+                    <h3 class="text-lg leading-6 font-medium text-gray-900">
```
Comment (slashplusweb) [inline resources/js/Pages/Posts/Index.vue]: As described in Create.vue

File: resources/js/Pages/Posts/Index.vue [hunk 4]
```diff
@@ -1,18 +1,81 @@
...
+                        Posts
+                    </h3>
+                </div>
+                <div class="ml-4 mt-2 flex-shrink-0">
+      <span class="inline-flex rounded-md shadow-sm">
+        <a :href="route('cms.posts.create')"
+           class="relative inline-flex items-center px-4 py-2 border border-transparent text-sm leading-5 font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-500 focus:outline-none focus:shadow-outline">
+          Create new post
+        </a>
+      </span>
```
Comment (slashplusweb) [inline resources/js/Pages/Posts/Index.vue]: Style: indentation

File: resources/js/Pages/Posts/Index.vue [hunk 5]
```diff
@@ -1,18 +1,81 @@
...
+                </div>
+            </div>
+        </div>
+        <div class="bg-white shadow overflow-hidden sm:rounded-md">
+            <ul>
+                <li v-for="post in posts">
+                    <a :href="route('cms.posts.show', {post: post.id})"
```
Comment (slashplusweb) [inline resources/js/Pages/Posts/Index.vue]: why not `inertia-link` tag?

File: resources/js/Pages/Posts/Show.vue [hunk 6]
```diff
@@ -1,15 +1,53 @@
 <template>
-    <layout title="posts">
-        <h1>posts index</h1>
+    <layout title="Show post">
+        <div>
+            <div class="md:flex md:items-center md:justify-between">
+                <div class="flex-1 min-w-0">
+                    <h3 class="text-lg leading-6 font-medium text-gray-900">
```
Comment (slashplusweb) [inline resources/js/Pages/Posts/Show.vue]: As described in Create.vue

File: resources/js/components/TagInput.vue [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                this.$emit("input", this.tags.map(tag => tag.text));
+            }
+        }
+    };
+</script>
+
+<style>
+    .ti-input{
+        border-radius: 0.25rem;
+    }
```
Comment (slashplusweb) [inline resources/js/components/TagInput.vue]: What is this for?

File: src/Http/routes.php [hunk 8]
```diff
@@ -7,7 +7,12 @@
 use Bambamboole\LaravelCms\Http\Controllers\UsersController;
 use Illuminate\Support\Facades\Route;
 
+Route::post('/posts', [PostsController::class, 'store'])->name('posts.store');
 Route::get('/posts', [PostsController::class, 'index'])->name('posts.index');
+Route::get('/posts/create', [PostsController::class, 'create'])->name('posts.create');
+Route::get('/posts/{post}', [PostsController::class, 'show'])->name('posts.show');
+Route::put('/posts/{post}', [PostsController::class, 'update'])->name('posts.update');
+Route::get('/posts/{post}/edit', [PostsController::class, 'edit'])->name('posts.edit');
```
Comment (slashplusweb) [inline src/Http/routes.php]: Maybe add the delete route as well?

File: resources/js/components/Fields/MarkdownField.vue [hunk 9]
```diff
@@ -0,0 +1,60 @@
+<template>
+    <div class="markdown-field">
+        <textarea id="editor"></textarea>
+    </div>
+</template>
+<script>
+    let CodeMirror = require('codemirror');
```
Comment (slashplusweb) [inline resources/js/components/Fields/MarkdownField.vue]: I would suggest adding another branch and PR for the complete markdown editor stuff.

File: src/Http/routes.php (hunk 8 above)
Comment (bambamboole) [reply src/Http/routes.php]: I will implement this afterwards :-)

**Pull request merged**